netsniff-ng.c

```diff
@@ -851,8 +860,15 @@ static int begin_multi_pcap_file(struct ctx *ctx)
 	if (ctx->device_out[strlen(ctx->device_out) - 1] == '/')
 		ctx->device_out[strlen(ctx->device_out) - 1] = 0;
 
-	slprintf(fname, sizeof(fname), "%s/%s%lu.pcap", ctx->device_out,
-		 ctx->prefix ? : "dump-", time(NULL));
+    if (ctx->overwrite_interval > 0) {
+        slprintf(fname, sizeof(fname), "%s/%s%010lu.pcap", ctx->device_out,
+                 ctx->prefix ? : "dump-", ctx->file_number++);
+        ctx->file_number = (ctx->file_number < ctx->overwrite_interval) ? ctx->file_number : 0;
+    }
+    else {
+        slprintf(fname, sizeof(fname), "%s/%s%lu.pcap", ctx->device_out,
+             ctx->prefix ? : "dump-", time(NULL));
+    }
```

Comment:
This block is almost the same code as in `next_multi_pcap_file` above. Could you try to refactor into a simpler helper function to avoid code duplication?